From ca542167580f3cf714890db2dc0933b3ba12e070 Mon Sep 17 00:00:00 2001
From: Mika Laitio <mika.laitio@amd.com>
Date: Tue, 3 Jun 2025 13:52:27 -0700
Subject: [PATCH 16/16] Dependencies cmake pytorch audio build fix

- options like no-shift-count-negative are not
  supported cl.exe from msvc and causes build
  errors on windows if cl.exe used
- options like /EHsc are not available on clang/clang++
  and causes build errors in windows if clang is used

Signed-off-by: Mika Laitio <mika.laitio@amd.com>
---
 cmake/Dependencies.cmake | 10 ++++++----
 cmake/public/utils.cmake |  9 +++++++--
 2 files changed, 13 insertions(+), 6 deletions(-)

diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index 4f6b2e4..7b0c677 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -1052,9 +1052,11 @@ if(USE_ROCM)
     list(APPEND HIP_CXX_FLAGS -D__HIP_NO_HALF_OPERATORS__=1)
     list(APPEND HIP_CXX_FLAGS -D__HIP_NO_HALF_CONVERSIONS__=1)
     list(APPEND HIP_CXX_FLAGS -DTORCH_HIP_VERSION=${TORCH_HIP_VERSION})
-    list(APPEND HIP_CXX_FLAGS -Wno-shift-count-negative)
-    list(APPEND HIP_CXX_FLAGS -Wno-shift-count-overflow)
-    list(APPEND HIP_CXX_FLAGS -Wno-duplicate-decl-specifier)
+    if(NOT WIN32)
+      list(APPEND HIP_CXX_FLAGS -Wno-shift-count-negative)
+      list(APPEND HIP_CXX_FLAGS -Wno-shift-count-overflow)
+      list(APPEND HIP_CXX_FLAGS -Wno-duplicate-decl-specifier)
+    endif()
     list(APPEND HIP_CXX_FLAGS -DCAFFE2_USE_MIOPEN)
     list(APPEND HIP_CXX_FLAGS -DTHRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_HIP)
     list(APPEND HIP_CXX_FLAGS -std=c++17)
@@ -1067,7 +1069,7 @@ if(USE_ROCM)
       add_definitions(-DROCM_ON_WINDOWS)
       list(APPEND HIP_CXX_FLAGS -fms-extensions)
       # Suppress warnings about dllexport.
-      list(APPEND HIP_CXX_FLAGS -Wno-ignored-attributes)
+      #list(APPEND HIP_CXX_FLAGS -Wno-ignored-attributes)
     endif()
     add_definitions(-DROCM_VERSION=${ROCM_VERSION_DEV_INT})
     add_definitions(-DTORCH_HIP_VERSION=${TORCH_HIP_VERSION})
diff --git a/cmake/public/utils.cmake b/cmake/public/utils.cmake
index 781a4e6..d86b80f 100644
--- a/cmake/public/utils.cmake
+++ b/cmake/public/utils.cmake
@@ -352,9 +352,14 @@ endmacro()
 function(torch_compile_options libname)
   set_property(TARGET ${libname} PROPERTY CXX_STANDARD 17)
 
+  # Do not set cl-compiler option on windows if Clang or GNU compiler is used
+  if(NOT ${CMAKE_CXX_COMPILER_ID} MATCHES "Clang|GNU")
+    if (MSVC)
+      option(USE_MSVC_OPTIONS "Use msvc compiler options" ON)
+    endif()
+  endif()
   # until they can be unified, keep these lists synced with setup.py
-  if(MSVC)
-
+  if (USE_MSVC_OPTIONS)
     if(MSVC_Z7_OVERRIDE)
       set(MSVC_DEBINFO_OPTION "/Z7")
     else()
-- 
2.43.0

